File: 1537-maximum-score-after-splitting-a-string/solution.java
class Solution {
    public int ones(String s){
        int c=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='1')c++;
        }
        return c;
    }
    public int zero(String s){
        int c=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='0'){
                c++;
            }
        }
        return c;
    }
    public int maxScore(String s) {
        int ans=0;
        for(int i=1;i<s.length();i++){
            int c=zero(s.substring(0,i));
            c+=ones(s.substring(i,s.length()));
            ans=Math.max(ans,c);
        }
        return ans;
    }
}

File: 1537-maximum-score-after-splitting-a-string/solution.cpp
class Solution {
public:
    int maxScore(string s) {
        int ans=0;
        for(char c : s){
            if(c=='0')ans++;
        }
        int zero=0;
        int maxs=0,n=s.length();
        for(int i=1;i<n;i++){
            if(s[i-1]=='0'){
                zero++;
                ans--;
            }
            int leftsum=zero;
            int rightsum=(n-i-ans);
            maxs=std::max(maxs,leftsum+rightsum);
        }
        return maxs;
    }
};
